project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java

```diff
@@ -464,6 +464,22 @@ public void testOverrideBuildOptions() {
         Assert.assertFalse(project.buildOptions().testReport());
     }
 
+    @Test(description = "tests overriding build options when editing Toml")
+    public void testOverrideBuildOptionsOnTomlEdit() {
+        Path projectPath = RESOURCE_DIRECTORY.resolve("projectWithBuildOptions");
+        // 1) Initialize the project instance
+        BuildProject project = null;
+        BuildOptions buildOptions = new BuildOptionsBuilder().offline(true).build();
+        try {
+            project = BuildProject.load(projectPath, buildOptions);
+        } catch (Exception e) {
+            Assert.fail(e.getMessage());
+        }
+        BallerinaToml newBallerinaToml = project.currentPackage().ballerinaToml().get().modify().apply();
+        Package newPackage = newBallerinaToml.packageInstance();
+        Assert.assertTrue(newPackage.project().buildOptions().offlineBuild());
```

Comment:
Shall we check further by adding / removing buildOptions in ballerinaToml

Reply:
The updated flag value will be set. Added test 4b5e84d797711f74a767c7a0f1f003b7596890ff